### PR TITLE
avoid pull behavior check for namespace pause image

### PR DESCRIPTION
**Pull request by sharanyad**
### Summary
https://github.com/aws/amazon-ecs-agent/issues/1837 

when `ECS_IMAGE_PULL_BEHAVIOR` is set to `once` and pause images are needed at the same time, agent will try to download pause image once and fail. this leads to the task failing. 

### Implementation details
avoid pull behavior check for pause image used for container namespaces setup 
### Testing

New tests cover the changes: 

Manual testing: 
set`ECS_IMAGE_PULL_BEHAVIOR=once` and use `pidMode:"task"` in task definition. 
Task fails to run without this code change and passes otherwise. 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/docker_task_engine.go [hunk 1]
```diff
@@ -670,6 +670,10 @@ func (engine *DockerTaskEngine) pullContainer(task *apitask.Task, container *api
 	case apicontainer.ContainerCNIPause:
 		// ContainerCNIPause image are managed at startup
 		return dockerapi.DockerContainerMetadata{}
+
+	case apicontainer.ContainerNamespacePause:
```
Comment (petderek) [inline agent/engine/docker_task_engine.go]: Wait, can you put this on one line?

`case api.ContainerCNIPause | api.ContainerNamespacePause:`

Comment (sharanyad) [reply agent/engine/docker_task_engine.go]: done

Comment (petderek) [reply agent/engine/docker_task_engine.go]: 🤦 oops

use ',' not '|' .... '|' does entirely the wrong thing

**Comment by sharanyad** [comment]
functional tests are failing due to low memory limit

**Pull request merged**